### PR TITLE
TC05

### PR DESCRIPTION
**Pull request by trangnhomanhcoi2601**

### CODE REVIEW

File: src/main/java/page_objects/LoginPage.java [hunk 1]
```diff
@@ -47,4 +47,14 @@ public void login(String user, String password) {
         getTxtPassword().sendKeys(password);
         getBtnLogin().click();
     }
+
+    public void repeatLogin(String user, String password) {
+        for (int i=0; i<4; i++) {
```
Comment (anhlvt) [inline src/main/java/page_objects/LoginPage.java]: should not hard code here

File: src/test/java/tests/LoginTests.java [hunk 2]
```diff
@@ -41,4 +42,13 @@ public void tc03_LoginWithInvalidPassword() {
         LogHelper.info("Check the error message");
         Assert.assertEquals(loginPage.getErrorLoginMessage(), "Invalid username or password. Please try again.", "Error login message doesn't display correctly");
     }
+
+    @Test(description = "System shows message when user enters wrong password 4 times")
+    public void tc05_LoginWithEnteringWrongPasswordSeveralTimes(){
+        LogHelper.info("Login with wrong password 4 times ");
+        loginPage.repeatLogin(Constants.USER, DataHelper.getRandomPassword());
```
Comment (anhlvt) [inline src/test/java/tests/LoginTests.java]: no need random invalid password

File: src/main/java/helpers/DataHelper.java [hunk 3]
```diff
@@ -1,4 +1,24 @@
...
+import com.github.javafaker.Faker;
+import java.util.Random;
+
 public class DataHelper {
+
+    private static final Faker faker = new Faker();
+    private static Random randomGenerator = new Random();
+
+    public static String getRandomEmail() {
+        int randomInt = randomGenerator.nextInt(1000);
```
Comment (anhlvt) [inline src/main/java/helpers/DataHelper.java]: Why don't you use Faker for all

File: src/test/java/tests/LoginTests.java (hunk 2 above)
Comment (trangnhomanhcoi2601) [reply src/test/java/tests/LoginTests.java]: I've already fixed it. Please help me check it. Thanks so much

File: src/main/java/helpers/DataHelper.java (hunk 3 above)
Comment (trangnhomanhcoi2601) [reply src/main/java/helpers/DataHelper.java]: I re-synced everything.

File: src/main/java/page_objects/LoginPage.java (hunk 1 above)
Comment (trangnhomanhcoi2601) [reply src/main/java/page_objects/LoginPage.java]: I fixed it

File: src/main/java/page_objects/LoginPage.java [hunk 4]
```diff
@@ -47,4 +47,14 @@ public void login(String user, String password) {
         getTxtPassword().sendKeys(password);
         getBtnLogin().click();
     }
+
+    public void repeatLogin(String user, String password, int numberOfTimes) {
+        for (int i=0; i<numberOfTimes; i++) {
```
Comment (anhlvt) [inline src/main/java/page_objects/LoginPage.java]: format code

Comment (trangnhomanhcoi2601) [reply src/main/java/page_objects/LoginPage.java]: I formatted code

File: src/main/java/helpers/DataHelper.java [hunk 5]
```diff
@@ -1,4 +1,22 @@
 package helpers;
 
+import com.github.javafaker.Faker;
+import java.util.Random;
+
 public class DataHelper {
+
+    private static final Faker faker = new Faker();
+    private static Random randomGenerator = new Random();
```
Comment (anhlvt) [inline src/main/java/helpers/DataHelper.java]: remove if not use

Comment (trangnhomanhcoi2601) [reply src/main/java/helpers/DataHelper.java]: I've already removed it

**Review by anhlvt — APPROVED** [review]

**Pull request merged**